_episodes/24_conditional-workflow.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Conditional workflows"
+teaching: 24
+exercises: 0
+questions:
+- "How do you write workflows with conditions?"
+objectives:
+- "Learn how to construct workflows containing conditional steps."
+keypoints:
+- ""
+---
+This workflow contains a conditional step and is executed based on the input. 
+This allows workflows to skip additional steps based on input parameters given at the start of the program or by previous steps.
+
+*conditional-workflow.cwl*
+
+<!-- 
+~~~
+{% include cwl/conditional-workflow/conditional-workflow.cwl %}
+~~~
+ -->
+
+```
+class: Workflow
+cwlVersion: v1.2
+inputs:
+  val: int
+
+steps:
+
+  step1:
+    in:
+      in1: val
+      a_new_var: val
+    run: foo.cwl
+    when: $(inputs.in1 < 1)
+    out: [out1]
+
+  step2:
+    in:
+      in1: val
+      a_new_var: val
+    run: foo.cwl
+    when: $(inputs.a_new_var > 2)
+    out: [out1]
+
+outputs:
+  out1:
+    type: string
+    outputSource:
+      - step1/out1
+      - step2/out1
+    pickValue: first_non_null
+
+requirements:
+  InlineJavascriptRequirement: {}
+  MultipleInputFeatureRequirement: {}
+```
+
+The first thing you'll notice is that this workflow is only compatible for cwl 1.2+
+
+```
+class: Workflow
+cwlVersion: v1.2
+```
+
+The first step of the worklfow (step1) contains two input properties and will execute foo.cwl when the conditions are met. The new property `when` is where the condition validation takes place. In this case only when `in1`  from the workflow contains a value `< 1` this step will be executed. 
+
+```
+steps:
+
+  step1:
+    in:
+      in1: val
+      a_new_var: val
+    run: foo.cwl
+    when: $(inputs.in1 < 1)
+    out: [out1]
+```
+
+Using the following command `cwltool cond-wf-003.1.cwl --val 0` the value will pass the first conditional step and will therefor be executed and is shown in the log by `INFO [step step1] start` whereas the second step is skipped as indicated by `INFO [step step2] will be skipped`.
```

Comment:
Worthwhile to note here that we can have variables in the expression that are not present in the underlying process. In this case it is `a_new_var`

Reply:
Might be more instructive to actually use `a_new_var` in the expression as well or instead of `in1`